cinema/views.py

```diff
@@ -34,25 +37,84 @@
     queryset = Movie.objects.all()
     serializer_class = MovieSerializer
 
+    @staticmethod
+    def _params_to_ints(query_string):
+        return [int(str_id) for str_id in query_string.split(",")]
+
     def get_serializer_class(self):
         if self.action == "list":
             return MovieListSerializer
 
         if self.action == "retrieve":
             return MovieDetailSerializer
 
-        return MovieSerializer
+        return self.serializer_class
+
+    def get_queryset(self):
+        queryset = self.queryset
+
+        actors = self.request.query_params.get("actors")
+        genres = self.request.query_params.get("genres")
+        title = self.request.query_params.get("title")
+        date = self.request.query_params.get("date")
+        if actors:
+            actors = self._params_to_ints(actors)
+            queryset = queryset.filter(actors__id__in=actors)
+        if genres:
+            genres = self._params_to_ints(genres)
+            queryset = queryset.filter(genres__in=genres)
+        if title:
+            queryset = queryset.filter(title__icontains=title)
+        if date:
+            queryset = queryset.filter(date=date)
+
+        return queryset.distinct()
 
 
 class MovieSessionViewSet(viewsets.ModelViewSet):
     queryset = MovieSession.objects.all()
     serializer_class = MovieSessionSerializer
 
+    def get_queryset(self):
+        queryset = self.queryset
+
+        movie = self.request.query_params.get("movie")
+        date = self.request.query_params.get("date")
+        if movie:
+            queryset = queryset.filter(movie_id=movie)
+        if date:
+            queryset = queryset.filter(show_time__date=date)
+        return queryset.distinct()
+
     def get_serializer_class(self):
         if self.action == "list":
             return MovieSessionListSerializer
 
         if self.action == "retrieve":
             return MovieSessionDetailSerializer
 
-        return MovieSessionSerializer
+        return self.serializer_class
+
+
+class OrderPagination(PageNumberPagination):
+    page_size = 3
+    page_size_query_param = "page"
```

Comment:
The `page_size_query_param` is currently set to 'page', which can be confusing as it might be mistaken for the page number parameter. Consider renaming it to 'page_size' to clearly indicate its purpose.